UM/Settings/Models/SettingDefinitionsModel.py

```diff
@@ -505,16 +501,33 @@ def forceUpdate(self) -> None:
     # Note that this triggers a model reset and should only be called when the
     # underlying data needs to be updated. Otherwise call _updateVisibleRows.
     def _update(self) -> None:
-        if not self._container:
+        if not self._stack:
+            self._definition_dict.clear()
+            self._definition_list.clear()
             return
 
         # Try and find a translation catalog for the definition
-        for file_name in self._container.getInheritedFiles():
+        for file_name in self._stack.getBottom().getInheritedFiles():
             catalog = i18nCatalog(os.path.basename(file_name))
             if catalog.hasTranslationLoaded():
                 self._i18n_catalog = catalog
 
-        new_definitions = self._container.findDefinitions()
+        # Terminology: if <my_def> inherits <base_def>, <base_def> is called the base definition and <my_def> is called
+        # the sub-definition.
+        # Get all definitions grouped by key recursively from bottom up, so if a same definition key is found in a
+        # sub-definition and a base definition, the one from the sub-definition will be used.
+        all_definitions_dict = {}
+        def createStackList(definition_dict, s):
+            if s is None:
+                return
+            createStackList(definition_dict, s.getNextStack())
+            definitions = s.getBottom().findDefinitions()
+            for d in definitions:
+                definition_dict[d.key] = d
+        createStackList(all_definitions_dict, self._stack)
```

Comment:
Why are you defining this function here and immediately using it once and throwing it away? It would be more clear with just a block of code, in my opinion.

Reply:
This is a recursion, so a function is needed...